gips/data/smap/smap.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python
+################################################################################
+#    SMAP: Soil Moisture Active Passive
+#
+#    AUTHOR: Subit Chakrabarti
+#    EMAIL:  chakrabarti.subit@gmail.com
+################################################################################
+
+from __future__ import print_function
+
+import os
+import sys
+import re
+import datetime
+
+import urllib
+import urllib2
+
+import math
+import numpy as np
+import requests
+
+import gippy
+from gippy.algorithms import Indices
+from gips.data.core import Repository, Asset, Data
+from gips.utils import VerboseOut, settings
+from gips import utils
+
+
+class smapRepository(Repository):
+    name = 'SMAP'
+    description = 'Soil Moisture Active Passive (SMAP)'
+
+    # NASA assets require special authentication
+    _manager_url = "https://urs.earthdata.nasa.gov"
+    _tile_attribute = 'tileid'
+
+class smapAsset(Asset):
+    Repository = smapRepository
+    _sensors = {'RAD': {'description': 'Soil Moisture Active Passive Radiometer'}}
+    _assets = {
+        'SM_P_E': {
+            'url': 'https://n5eil01u.ecs.nsidc.org/SMAP/SPL3SMP_E.002',
+            'pattern': r'SMAP\_.{2}\_SM\_P\_E\_.{8}\_.{6}\_.{3}\.h5',
+            'description': 'Passive Enhanced Radiometer Based SM at 9KM',
+            'startdate': datetime.date(2015, 3, 31),
+            'latency': 1,
+        },
+        'SM_P': {
+            'url': 'https://n5eil01u.ecs.nsidc.org/SMAP/SPL3SMP.005',
+            'pattern': r'SMAP\_.{2}\_SM\_P\_.{8}\_.{6}\_.{3}\.h5',
+            'description': 'Passive Original Radiometer Based SM at 36KM',
+            'startdate': datetime.date(2015, 3, 31),
+            'latency': 1,
+        },
+    }
+
+    def __init__(self, filename):
+        """ Inspect a single file and get some metadata """
+        super(smapAsset, self).__init__(filename)
+
+        bname = os.path.basename(filename)
+        self.asset = (re.search('(?<=SMAP_L3_)\w*(?=_[0-9]{8})', bname)).group(0)
+        date_here = (re.search('[0-9]{8}', bname)).group(0)
+        self.date = datetime.datetime.strptime(date_here, "%Y%m%d").date()
+        self._version = (re.search('R[0-9]*', bname)).group(0)
+        self.tile = 'h01v01'
+
+    @classmethod
+    def query_provider(cls, asset, tile, date):
```

Comment:
Similarly with previous code, I'm unsure that cutpasting from modis was the right way to do this.  We generally prefer to refactor into common code that multiple sources then call into.  Was that considered?  How was the decision made?